R/call.R

```diff
@@ -1050,3 +1050,24 @@ call_type <- function(x) {
     abort("corrupt language object")
   }
 }
+
+call_zap_inline <- function(x) {
+  if (is_call(x)) {
+    if (is_call(x, "function")) {
+      if (!is_null(x[[2]])) {
+        x[[2]] <- as.pairlist(map(x[[2]], call_zap_inline))
+      }
+      x[[3]] <- call_zap_inline(x[[3]])
+    } else {
+      x[] <- map(x, call_zap_inline)
+    }
+    return(x)
+  }
+
+  if (is_symbol(x) || is_syntactic_literal(x)) {
+    x
```

Comment:
I find this slightly easier to understand, but it's not that important

```suggestion
    x
  } else if (is_symbol(x) || is_syntactic_literal(x)) {
    x
```

Reply:
The idea here is to keep the terminating and recursive branches separate.